stan/math/opencl/copy.hpp

```diff
@@ -147,23 +168,24 @@ inline std::vector<T> packed_copy(const matrix_cl<T>& src) {
  * size of the vector does not match the expected size
  * for the packed triangular matrix
  */
-template <matrix_cl_view matrix_view, typename T,
-          typename = require_arithmetic_t<T>>
-inline matrix_cl<T> packed_copy(const std::vector<T>& src, int rows) {
+template <matrix_cl_view matrix_view, typename Vec,
+          typename Vec_scalar = scalar_type_t<Vec>,
+          require_std_vector_t<Vec>...>
+inline matrix_cl<Vec_scalar> packed_copy(Vec&& src, int rows) {
   const int packed_size = rows * (rows + 1) / 2;
   check_size_match("copy (packed std::vector -> OpenCL)", "src.size()",
                    src.size(), "rows * (rows + 1) / 2", packed_size);
-  matrix_cl<T> dst(rows, rows, matrix_view);
+  matrix_cl<Vec_scalar> dst(rows, rows, matrix_view);
   if (dst.size() == 0) {
     return dst;
   }
   try {
-    matrix_cl<T> packed(packed_size, 1);
+    matrix_cl<Vec_scalar> packed(packed_size, 1);
     cl::Event packed_event;
     const cl::CommandQueue queue = opencl_context.queue();
-    queue.enqueueWriteBuffer(packed.buffer(), CL_FALSE, 0,
-                             sizeof(T) * packed_size, src.data(), nullptr,
-                             &packed_event);
+    queue.enqueueWriteBuffer(packed.buffer(), opencl_context.in_order(), 0,
```

Comment:
```suggestion
    queue.enqueueWriteBuffer(packed.buffer(), std::is_rvalue_reference<Vec&&> || opencl_context.in_order(), 0,
```
Everything is prepared, but the actual bug is not fixed in this function.